htdocs/install/mysql/tables/llx_user_perentity-multicompany.sql

```diff
@@ -0,0 +1,33 @@
+-- ============================================================================
+-- Copyright (C) 2010-2021 Regis Houssin  <regis.houssin@inodbox.com>
+--
+-- This program is free software; you can redistribute it and/or modify
+-- it under the terms of the GNU General Public License as published by
+-- the Free Software Foundation; either version 3 of the License, or
+-- (at your option) any later version.
+--
+-- This program is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU General Public License for more details.
+--
+-- You should have received a copy of the GNU General Public License
+-- along with this program; if not, write to the Free Software
+-- Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+--
+-- ===========================================================================
+
+create table llx_user_perentity
```

Comment:
Currently a user and its properties are already linked to 1 entity. Also permission system are also on a user. Having one user with different properties for n entities (what you introduce) seems not compatible with n user for one entity (what we need for permission system and security to work).
Adding such a feature will bring a very high complexity and a lot of troubles into the core, some without solution, for the need of 1 external module.
For the moment the permission system for multicompany is still not clear and is source of a lot of trouble to know which data can be seen or not. I prefer to not add more complexity on users management in the core and let the trouble this will generate out of Dolibarr core, so inside external modules.

Reply:
@eldy 
I wanted to include it in the core like llx_product_perentity and llx_societe_perentity so that it is fully integrated, for Dolibarr and for external modules that support Multicompany. Otherwise it is unmanageable for the moment, because there is too much code that we cannot customize and we would have inconsistencies.